### PR TITLE
enable github actions ci to build the project

### PR DESCRIPTION
**Pull request by chengfang**

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      if: contains(matrix.os, 'ubuntu')
+      run: |
+        sudo bash -c "echo '127.0.0.1   localhost localhost.localdomain localhost4 localhost4.localdomain4' > /etc/hosts"
+        sudo bash -c "echo '::1         localhost localhost.localdomain localhost6 localhost6.localdomain6' >> /etc/hosts"
+    - name: Configure Runner - Windows
+      if: contains(matrix.os, 'windows')
+      run: |
+        echo '127.0.0.1   localhost localhost.localdomain localhost4 localhost4.localdomain4' > %SystemRoot%\System32\drivers\etc\hosts
+        echo '::1         localhost localhost.localdomain localhost6 localhost6.localdomain6' >> %SystemRoot%\System32\drivers\etc\hosts
+      shell: cmd
```
Comment (bartoszmajsak) [inline .github/workflows/build.yml:35]: Could you add a comment explaining why exactly is the host mapping needed?

Comment (chengfang) [reply .github/workflows/build.yml]: According to Ken from WildFly, the default hosts file in GitHub has a lot of hosts mapped to localhost (spam sites, Bitcoin mining sites etc) and it caused some weird behaviour in some tests.  gethostbyname("localhost") can return a random entry from /etc/hosts. That's why these steps are needed in building and running tests in WildFly. But I think it's less of a concern for this project, so I've removed these steps to simplify it a bit.

**Comment by WolfgangHG** [comment]
Never have worked with CI, so a beginners question: will those tests use any profiles? It might make sense to test for all profiles for managed servers.

**Comment by chengfang** [comment]
@WolfgangHG yes, we can enhance it by adding more jobs for different profiles, to make it easier to see which profile tests failed. I expect these enhancement to follow this basic setup.

**Comment by chengfang** [comment]
Okay, I'll go ahead merge it and see how it goes.

**Pull request merged**